src/lambda/commands/createNewSamApp.ts

```diff
@@ -55,43 +54,47 @@ export async function resumeCreateNewSamApp(activationLaunchPath: ActivationLaun
 }
 
 export interface CreateNewSamApplicationResults {
-    reason: 'unknown' | 'userCancelled' | 'fileNotFound' | 'complete' | 'error'
-    result: 'pass' | 'fail' | 'cancel'
     runtime: string
+    result: result
 }
 
+type createReason = 'unknown' | 'userCancelled' | 'fileNotFound' | 'complete' | 'error'
+
 /**
  * Runs `sam init` in the given context and returns useful metadata about its invocation
  */
 export async function createNewSamApplication(
     channelLogger: ChannelLogger,
     samCliContext: SamCliContext = getSamCliContext(),
     activationLaunchPath: ActivationLaunchPath = new ActivationLaunchPath()
-): Promise<CreateNewSamApplicationResults> {
-    const results: CreateNewSamApplicationResults = {
-        reason: 'unknown',
-        result: 'fail',
-        runtime: 'unknown'
-    }
+): Promise<void> {
+    let createResult: result = 'Succeeded'
+    let reason: createReason = 'unknown'
+    let createRuntime: runtime = 'unknown'
+    let config: CreateNewSamAppWizardResponse | undefined
+
+    let initArguments: SamCliInitArgs
 
     try {
         await validateSamCli(samCliContext.validator)
 
         const wizardContext = new DefaultCreateNewSamAppWizardContext()
-        const config: CreateNewSamAppWizardResponse | undefined = await new CreateNewSamAppWizard(wizardContext).run()
+        config = await new CreateNewSamAppWizard(wizardContext).run()
         if (!config) {
-            results.result = 'cancel'
-            results.reason = 'userCancelled'
+            createResult = 'Cancelled'
+            reason = 'userCancelled'
 
-            return results
+            return
         }
 
-        results.runtime = config.runtime
+        // This cast (and all like it) will always succeed because Runtime (from config.runtime) is the same
+        // section of types as runtime
+        createRuntime = config.runtime as runtime
```

Comment:
Can we swap `config.runtime` to use the `runtime` type?

Reply:
We could but I'd prefer "duplication" of types over relying on telemetry types in core features. Although they are "the same" they mean different things